src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -578,7 +580,12 @@ private BuildEventContext ReadBuildEventContext()
             int taskId = ReadInt32();
             int submissionId = ReadInt32();
             int projectInstanceId = ReadInt32();
-            int evaluationId = ReadInt32();
+
+            // evaluationId was introduced in format version 2
+            if (fileFormatVersion > 1)
+            {
+                int evaluationId = ReadInt32();
```

Comment:
BuildEventContext has a constructor without the evaluationID but all the rest. Or just initialize the id with BuildEventContext.InvalidEvaluationId

Reply:
LOL this is how we know it's been a long Friday :)